pkg/daemon/file_writers.go

```diff
@@ -45,12 +46,50 @@ func noOrigFileStampName(fpath string) string {
 	return filepath.Join(noOrigParentDir(), fpath+".mcdnoorig")
 }
 
-func createOrigFile(fromPath, fpath string) error {
-	if _, err := os.Stat(noOrigFileStampName(fpath)); err == nil {
-		// we already created the no orig file for this default file
-		return nil
+func createOrigFile(fromPath, fpath string, isCoreOSVariant, forceCreate bool) error {
+
+	// Declare a boolean variable - orig - to indicate whether an orig file should be created
+	// for the given path
+	// true - orig file
+	// false - noOrig file
+	var orig bool
+
+	// https://issues.redhat.com/browse/OCPBUGS-11437
+	// MCO keeps the pull secret to .orig file once it replaced
+	// Adapt a check used in function deleteStaleData for backwards compatibility: basically if the file doesn't
+	// exist in /usr/etc (on FCOS/RHCOS) and no rpm is claiming it, we assume that the orig file came from a
+	// wrongful backup of a MachineConfig file instead of a file originally on disk.
+	if _, err := exec.Command("rpm", "-qf", fpath).CombinedOutput(); err == nil {
+		// File is owned by an rpm
+		orig = true
+	} else if strings.HasPrefix(fpath, "/etc") && isCoreOSVariant {
+		if _, err := os.Stat(withUsrPath(fpath)); err != nil {
+			if !os.IsNotExist(err) {
+				return err
+			}
+		} else {
+			orig = true
+		}
+	} else if forceCreate {
+		orig = true
+	} else {
+		orig = false
```

Comment:
nit: you can declare orig as false originally and skip this else condition (i.e. `var orig bool` -> `orig := false`